tree/tree/test/TChainParsing.cxx

```diff
@@ -68,3 +75,55 @@ TEST(TChainParsing, LocalAdd)
    EXPECT_STREQ(files->At(7)->GetTitle(), "/path/to/file.root");
    EXPECT_STREQ(files->At(7)->GetName(), "/treename");
 }
+
+void FillTree(const char *filename, const char *treeName)
+{
+   TFile f{filename, "RECREATE"};
```

Comment:
I recommend that you check for the file to be 'okay' (so not `IsZombie`) before proceeding and then give up early (rather than fail the test because the files don't exists).

Reply:
The test checks that `GetListOfFiles` returns an actual `TList *` and not a `nullptr`. I do not check that the files exist, because it's not the point of the test. In this test, I assume that `TFile::Close` will work properly. I can add another check for that nonetheless of course

Reply:
I'm noticing now that the comment is placed inside the `FillTree` function, which is not testing anything. It's just a helper function to create a file. The test starts from line 96 with `TEST(TChainParsing, GlobbingWithTreenameToken)`

Reply:
What I am mean is that essentially we should have :
```
   if ( FillTree(fileName.c_str(), "events") == failure_return_value) {
       Error("testname", "The test setup failed, we are not proceeding with the actual test.");
       return error_code;
  }
```
(for example if the current directory is not writeable (full disk)

Reply:
Reading the comment once more, I think I understand what you mean now. Opening the file may file, for whatever reason, and it's better to check that early and throw an error here, rather than wait to see it in the actual test, and wonder whether it's `TFile`'s fault  for not opening the file or `TChain`'s fault for not creating the correct list of files

Reply:
In the last commit I added a check after file creation. I wonder how to deal with the file being properly written and closed. Can I use `f.IsZombie` after the call to `f.Close` to check that?

Reply:
> Can I use f.IsZombie after the call to f.Close to check that?

Unfortunately, no.  There is no good indication (beside error message) that it failed.

The other test can that be added is testing that `Write` returned `>0`.

Reply:
Ok thanks! I also included that check. Let me know if the PR is ok to be merged